src/Features/OnboardingTasks/TaskLists.php

```diff
@@ -83,18 +110,15 @@ public static function add_task( $list_id, $args ) {
 	 * Add default task lists.
 	 */
 	public static function maybe_add_default_tasks() {
-		$added = isset( self::$lists['setup'] );
+		global $wp;
 
-		if ( ! apply_filters( 'woocommerce_admin_onboarding_tasks_add_default_tasks', ! $added ) ) {
+		if ( substr( $wp->request, 0, 34 ) !== 'wp-json/wc-admin/onboarding/tasks' ) {
```

Comment:
The WP REST API prefix [is filterable](https://github.com/WordPress/WordPress/blob/cf0274e1ca419f467ed46d60287a6dd9908f3bc0/wp-includes/rest-api.php#L395). I realize that this isn't the first hardcoded usage of it in WC, but we should be more careful.

Perhaps moving the call of `maybe_add_default_tasks()` to `src/API/OnboardingTasks.php` in the route callbacks that need it would make this more resilient?

Reply:
Thanks for the idea, @jeffstieler.  This is a temporary fix that was already merged to 2.8, but I'll keep this in mind going forward.

Ideally we won't only be adding via REST calls anyway since it reduces the capability of task list checks outside of the REST endpoints.